QModManager/API/ModLoading/Internal/QModPlaceholder.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace QModManager.API.ModLoading.Internal
+{
+    using System.Reflection;
+
+    internal class QModPlaceholder : QMod, IQMod
+    {
+        private readonly ModStatus status;
+        /// <summary>
+        /// The dummy <see cref="QMod"/> which is used to represent QModManager
+        /// </summary>
+        internal static QModPlaceholder QModManager { get; } = new QModPlaceholder
+        {
+            Id = "QModManager",
+            DisplayName = "QModManager",
+            Author = "The QModManager Dev Team",
+            LoadedAssembly = Assembly.GetExecutingAssembly(),
+            SupportedGame = QModGame.Both,
+        };
+
+        private QModPlaceholder()
+        {
+            // Hide empty constructor. Only to be used for QModManager singlton.
+            status = ModStatus.Success;
+        }
+
+        internal QModPlaceholder(string name)
+        {
+            this.Id = Patcher.IDRegex.Replace(name, "");
+            this.DisplayName = name;
+            this.Author = "Unknown";
+            this.SupportedGame = QModGame.None;
+            this.Enable = false;
```

Comment:
How does `Enable` work currently?

Reply:
Handled in `QModFactory`, this in intended to prevent the mod from patching.
```CSharp
if (!mod.Enable)
{
    earlyErrors.Add(mod, ModStatus.CanceledByUser);
    continue;
}
```
As for how new mods will change this value, that is still yet to be determined unless we leave it inside the `mod.json` and keep using that file (in a simplified form) to handle things like this.

Reply:
So we might just continue using `mod.json` for this. 
`SAM` and `Vortex` are likely both using this field.

I'll have to make some adjustments to the DLL loading so that it still checks the `mod.json` for that value.